packages/react-native-test-renderer/src/renderer/FabricUIManagerMock.js

```diff
@@ -0,0 +1,701 @@
+
+  function ensureHostNode(node: Node): void {
+    if (node == null || typeof node !== 'object') {
+      throw new Error(
+        `Expected node to be an object. Got ${
+          node === null ? 'null' : typeof node
+        } value`,
+      );
+    }
+
+    if (typeof node.viewName !== 'string') {
+      throw new Error(
+        `Expected node to be a host node. Got object with ${
+          node.viewName === null ? 'null' : typeof node.viewName
+        } viewName`,
+      );
+    }
+  }
+
+  function getAncestorsInChildSet(
+    node: Node,
+    childSet: NodeSet,
+  ): ?$ReadOnlyArray<[Node, number]> {
+    const rootNode = toNode({
+      reactTag: 0,
+      rootTag: fromNode(node).rootTag,
+      viewName: 'RootNode',
+      // $FlowExpectedError
+      instanceHandle: null,
+      props: {},
+      children: childSet,
+    });
+
+    let position = 0;
+    for (const child of childSet) {
+      const ancestors = getAncestors(child, node);
+      if (ancestors) {
+        return [[rootNode, position]].concat(ancestors);
+      }
+      position++;
+    }
+
+    return null;
+  }
+
+  function getAncestorsInCurrentTree(
+    node: Node,
+  ): ?$ReadOnlyArray<[Node, number]> {
+    const childSet = roots.get(fromNode(node).rootTag);
+    if (childSet == null) {
+      return null;
+    }
+
+    return getAncestorsInChildSet(node, childSet);
+  }
+
+  function getAncestors(root: Node, node: Node): ?$ReadOnlyArray<[Node, number]> {
+    if (fromNode(root).reactTag === fromNode(node).reactTag) {
+      return [];
+    }
+
+    let position = 0;
+    for (const child of fromNode(root).children) {
+      const ancestors = getAncestors(child, node);
+      if (ancestors != null) {
+        return [[root, position]].concat(ancestors);
+      }
+      position++;
+    }
+
+    return null;
+  }
+
+  export function getNodeInChildSet(node: Node, childSet: NodeSet): ?Node {
+    const ancestors = getAncestorsInChildSet(node, childSet);
+    if (ancestors == null) {
+      return null;
+    }
+
+    const [parent, position] = ancestors[ancestors.length - 1];
+    const nodeInCurrentTree = fromNode(parent).children[position];
+    return nodeInCurrentTree;
+  }
+
+  function getNodeInCurrentTree(node: Node): ?Node {
+    const childSet = roots.get(fromNode(node).rootTag);
+    if (childSet == null) {
+      return null;
+    }
+
+    return getNodeInChildSet(node, childSet);
+  }
+
+  function* dfs(node: ?Node): Iterator<Node> {
+    if (node == null) {
+      return;
+    }
+
+    yield node;
+
+    for (const child of fromNode(node).children) {
+      yield* dfs(child);
+    }
+  }
+
+  function hasDisplayNone(node: Node): boolean {
+    const props = fromNode(node).props;
+    // Style is flattened when passed to native, so there's no style object.
+    // $FlowFixMe[prop-missing]
+    return props != null && props.display === 'none';
+  }
+
+  interface IFabricUIManagerMock extends FabricUIManager {
+    fireEvent(
```

Comment:
[nit] Consider naming `dispatchEvent` to match browser JS naming: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent

Afaik, `fireEvent` is Testing Library specific name.